### PR TITLE
add -experimental to enable rescheduling

### PR DESCRIPTION
**Pull request by vieux**
Signed-off-by: Victor Vieux <vieux@docker.com>

### CODE REVIEW

**Comment by vieux** [comment]
ping @docker/swarm-maintainers what do you think of this ?

File: cluster/config.go [hunk 1]
```diff
@@ -86,25 +87,27 @@ func BuildContainerConfig(c dockerclient.ContainerConfig) *ContainerConfig {
...
-			affinities = append(affinities, value)
-		} else if ok && key == "constraint" {
-			constraints = append(constraints, value)
-		} else if ok && key == "reschedule" {
-			reschedulePolicies = append(reschedulePolicies, value)
-		} else {
-			env = append(env, e)
+		// parse affinities/constraints/reschedule policies from env (ex. docker run -e affinity:container==redis -e affinity:image==nginx -e constraint:region==us-east -e constraint:storage==ssd -e reschedule:off)
+		for _, e := range c.Env {
+			if ok, key, value := parseEnv(e); ok && key == "affinity" {
```
Comment (dongluochen) [inline cluster/config.go]: affinity and constraint are not experimental.

**Comment by abronan** [comment]
@docker/swarm-maintainers So the context is that I expressed my concerns for #1624 because I though that adding `experimental` on the restart flag was an issue when we'll change the flag back when it's out of experimental (for example users scripting their deployments, etc. this would be a breaking change).

With this change it's actually a flag on the Manager which I think is better because not breaking existing scripts and deployments.

File: cli/cli.go [hunk 2]
```diff
@@ -32,6 +33,11 @@ func Run() {
 			Value: "info",
 			Usage: fmt.Sprintf("Log level (options: debug, info, warn, error, fatal, panic)"),
 		},
+
+		cli.BoolFlag{
+			Name:  "experimental",
```
Comment (dongluochen) [inline cli/cli.go]: Why not use a flag like `autoreschdule`? You can state experimental in flag description. `experimental` is not one feature. Over time this flag would be used for other features. It's not backward compatible.

File: cluster/config.go (hunk 1 above)
Comment (vieux) [inline cluster/config.go]: right

File: cli/cli.go (hunk 2 above)
Comment (vieux) [inline cli/cli.go]: Yes that was my point,
docker engine for example has one regular build and one experimental build. not one build per experimental feature.

Going forward, ppl who wants to test the bleeding edge stuff just enable "-experimental" and that's it, they don't have to remove/add flags for every release.

**Comment by vieux** [comment]
I'll add `Experimental: true` in the Info API response once this one gets merged (if it does) because it required yet another godep update.

**Comment by vieux** [comment]
@abronan actually I put the flag on swarm itself, not the manager, so if one day we have experimental features on join on list, it will make things easier :)

**Comment by abronan** [comment]
@vieux even better imho :+1:

**Comment by dongluochen** [comment]
@vieux There is conflict please rebase.

**Comment by vieux** [comment]
@dongluochen done. sometime I'm disappointed by git, this was was a no brainer, it could have resolved it.

**Comment by dongluochen** [comment]
LGTM.

**Comment by aluzzardi** [comment]
LGTM

**Comment by moxiegirl** [comment]
docs LGTM

**Pull request merged**